sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/resourcemanager/AzureEventHubResourceManagerAutoConfiguration.java

```diff
@@ -33,7 +34,7 @@ public AzureEventHubResourceManagerAutoConfiguration(AzureResourceManager azureR
     @Bean
     @ConditionalOnMissingBean
     @ConditionalOnProperty("spring.cloud.azure.eventhub.namespace")
-    // TODO(xiada) conditional on missing connection-string property
+    @ConditionalOnExpression("T(org.springframework.util.StringUtils).isEmpty('${spring.cloud.azure.eventhub.connection-string:}')")
```

Comment:
won't this expression always true? since you have a ';' at the end of that value property

Reply:
Not if the connection-string is set.